### PR TITLE
Upgrade vagrant box to run on python 3.6

### LINKED ISSUE #1333: Upgrade Vagrant box to run on Python 3.6
We probably need to use a newer Ubuntu box in https://github.com/python/pythondotorg/blob/master/provisioning/pythondotorg.yml to install Python 3.6 from the official Ubuntu repositories.

Also, https://pythondotorg.readthedocs.io/install.html#easy-setup-using-vagrant documentation needs to be updated.

### PR DESCRIPTION
**Pull request by Tomasz-Kluczkowski**
Changed vagrant box to bento/ubuntu-18.04 which gives Python 3.6.5.

Fixes #1333

### CODE REVIEW

File: provisioning/pythondotorg.yml [hunk 1]
```diff
@@ -36,13 +31,6 @@
       changed_when:
         "result.stdout is defined and '0 upgraded, 0 newly installed, 0 to remove and' not in result.stdout"
 
-    - name: Install python-lxml deps
-      become: yes
-      apt: pkg=python-lxml state=build-dep
```
Comment (berkerpeksag) [inline provisioning/pythondotorg.yml:41]: Does this get installed by BeautifulSoup now?

**Review by berkerpeksag — COMMENTED** [review]
This looks good, but I haven't tested it locally yet. Could you also remove the commented-out line at line 29:

```yml
# - acl
```

File: docs/source/install.rst [hunk 2]
```diff
@@ -18,7 +18,7 @@ You should then be able to provision the Vagrant box.
 
     $ vagrant up
 
-The box will be provisioned with Python 3.5, a virtual environment with all
+The box will be provisioned with Python 3.6.5, a virtual environment with all
```
Comment (berkerpeksag) [inline docs/source/install.rst]: Nit: How about saying something like "the latest Python 3.6 release" or "version" so we don't need to update the version information every six month :)

File: provisioning/pythondotorg.yml (hunk 1 above)
Comment (Tomasz-Kluczkowski) [reply provisioning/pythondotorg.yml:41]: Sure I will remove # - acl in the next amendment.
Regading lxml:
I have done local tests yesterday during the Python Sprint of the new box and was able to launch the site with initial data.
Also built the docs from txt to html and run the management command to do the same.
It looks like lxml required works now without specifying those dependencies.
I will retest all of that though just to make sure. Is there anything else that lxml is used for?

File: docs/source/install.rst (hunk 2 above)
Comment (Tomasz-Kluczkowski) [reply docs/source/install.rst]: Ok, will do.
I need a day or two though as was really busy today - work / training :).

File: provisioning/pythondotorg.yml (hunk 1 above)
Comment (berkerpeksag) [reply provisioning/pythondotorg.yml:41]: > Is there anything else that lxml is used for?

I think BeautilfulSoup + lxml is only used by the PEP converter, so running `./manage.py test peps` may help to confirm this.

File: docs/source/install.rst (hunk 2 above)
Comment (berkerpeksag) [reply docs/source/install.rst]: Sure, there is no rush :)

File: provisioning/pythondotorg.yml (hunk 1 above)
Comment (Gautier) [reply provisioning/pythondotorg.yml:41]: `lxml` is now available as a wheel, so `pip install` does not need to build it any-more which means we don't need the `pkg=python-lxml state=build-dep` step and we can also remove `libxml2-dev` and `libxslt-dev` from the `Install base packages` step.

I have tested locally, `./manage.py test peps` passes and I can generate the PEPs.

Comment (berkerpeksag) [reply provisioning/pythondotorg.yml:41]: Awesome, thanks for testing @Gautier! @Tomasz-Kluczkowski could you remove those packages as well?

Comment (Tomasz-Kluczkowski) [reply provisioning/pythondotorg.yml:41]: Working on it.

Comment (Tomasz-Kluczkowski) [reply provisioning/pythondotorg.yml:41]: Added Gautier to the commit as a co-autor - he helped a lot during the sprint.

**Review by berkerpeksag — APPROVED** [review]

**Comment by berkerpeksag** [comment]
Thanks, @Tomasz-Kluczkowski and @Gautier!

**Pull request merged**